tests/integration_tests/cpu/propagator/jacobian_validation.cpp

```diff
@@ -1612,12 +1609,28 @@ int main(int argc, char** argv) {
                          decltype(rect_det)::masks::id::e_rectangle2>(
                 rect_det, 1u, helix_bz.dir(detector_length));
 
+        // With volume material
+        rectangle_cfg.volume_material(volume_mat);
+        auto [rect_det_w_mat, rect_names2] =
+            create_telescope_detector(host_mr, rectangle_cfg);
+        [[maybe_unused]] const auto [euler_rect_initial2, shift_rect_initial2] =
+            tilt_surface<decltype(rect_det_w_mat),
```

Comment:
Do we need this new lines of code?

In this test, the detector is meant to have the same rotation angles for the same track.
Now this PR is sampling the different angles for the new rectangular geometry, wire geometry and the original one.

Reply:
I have no idea, tbh. The tilt function seemed to add transforms to the detector. We need different detectors for the different volume materials, because a detector must not be changed after construction (the methods that make this currently possible will be gone soon)

Reply:
Is it impossible to call the material store and swap the material in it?

Reply:
I should be be, yes. The only reason it is not yet, is that the toy detector and wire chamber construction have not been refactored to use the detector builder.
This particular case may be trivial, but in general the data dependencies in the detector can be quite complex and allowing people to manipulate it (or change it by accident) can lead to all sorts of weird bugs. For example, if you try changing the material for one surface/volume, you may in fact be changing it for many that point to the same instance without noticing it

Reply:
Can you utilize what is in #689 to provide the same angles to different detectors

Reply:
I don't know the code and I am not entirely sure how you want me to change it exactly?

Reply:
I made tilt_surface to take Euler angles as arguments rather than generating inside the function.
Meaning that you can construct two different geometries but still can take the same Euler angles, which will solve my worry in https://github.com/acts-project/detray/pull/687#discussion_r1519010038

Reply:
Ok, nice. So I just have to rebase onto #689?

Reply:
I am fixing a bug right now so just copy and past what you need from there if you need to get it included relatively soon

Reply:
Rebased, please have a look